### ISSUE TITLE
Sync new litgpt changes

### ISSUE THREAD

**Issue by gabikadlecova** [whittle-org/whittle#222]
**Is your feature request related to a problem? Please describe.**
There were some recent changes in the litgpt codebase (e.g. in KV caching). We should make whittle up-to-date with the changes. Additionally, we should state with which version we're compatible/tested.

**Comment by gabikadlecova** [comment]
Note that 1. may not be as simple as I initially wrote - investigation is needed! It is not trivial to select QKV indices correctly

**Comment by aaronkl** [comment]
@rheasukthanker could you describe other necessary changes here?

**Comment by rheasukthanker** [comment]
Extended changes summary for base model: 
1. Add mask cache warning https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L70
2. Input pos warning https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L126
3. Changes to https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L647
4. Shape of cos, sin https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L133
5. More warnings https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L136
6. Changes in mask reshape https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L139
7. New arguments in forward https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L90
8. Add check for ```input_pos_maxp1``` https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L143
9. Change in block forward args https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L161
10. Head clamping https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L163
11. Change in head clamping and chunking logic https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L168
12. Minor changes in rope cache https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L193
13. Attention signature change https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L310
14. New attn_bias parameter in config https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L332
15. Changes in qkv splitting (check detailed summary above) https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L378. We might not need the ugly qkv indices anymore
16. QK norm https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L345 https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L387
17. input_pos_maxp1 in attention https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L415
18. expand->repeat_interleave in https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L425
19. Scores soft capping https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L468
20. Some changes in rope cache https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/model.py#L587. Irrelevant if we use tht function from litgpt directly. 
21.

**Comment by rheasukthanker** [comment]
LoRA changes 
1. Changes to loraqkv linear lora_ind (corresponding to changes in attention, qkv non-interleaved version https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/lora.py#L266)
2. LoRALinear-> create lora linear https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/lora.py#L490C24-L490C42
3. Changes in lora attention _load_from_state_dict https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/lora.py#L560

**Comment by rheasukthanker** [comment]
Pretraining
1. Compute token num https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/pretrain.py#L240
2. Logging https://github.com/Lightning-AI/litgpt/blob/v0.5.7/litgpt/pretrain.py#L253
